### ISSUE TITLE
Visual problems after Homey 2.0

### ISSUE THREAD

**Issue by r00p** [appzer/Athom-Homey-Pushsafer#5]
Hi!

Got some weird issues with this app in the new http://flow.athom.com

First the the problems that we have to fill in every field, even though we don't need all fields.
Isnt this possible to solve?

Second is that because the app have so many fields it gets totally bugged out on flow.athom.com.
the headlines also seems wrong. Many are called the same (Number,Text,text,number)

It is a very hard time manage this app in homey and i would really love if you could solve the basic issues. Low pushsafer because of the sound and many settings, but sometimes its a bit messy!

![pushs](https://user-images.githubusercontent.com/37667257/54091146-b00a9500-437c-11e9-9f8e-9abfeabc1ce5.JPG)

**Comment by appzer** [comment]
Hi, pushsafer module is based on an other i found. Other users test it for me, because i have no homey!
Is there a way to emulate the UI on a computer?

**Comment by Veldkornet** [comment]
I don’t think that either of these problems are a problem with the app.

There is no option to set a field as optional... this is how Homey works. If the field is in the card, it must be set.
Unless Athom change this, I suppose I would be possible to (as a workaround) make most of these settings available in the app settings which would then be considered default. 
The, make a “basic” flow card with only the minimal amount of fields, all others will be set from the defaults in the settings.
Then you could have an “advanced” card, where all of the options are available and would override the options set in the settings. 
Of course, it would be easier if the apps were given more flexibility. 

As for the view from the flow.athom.com site, I think that this is a problem with the site.
If you check the mobile app, all of the fields have example descriptions.
The fact that there are no actual titles, but only number, number, text, text, number etc is a bit annoying, but again, this is just how Homey works. 

It would probably be a better idea to contact Athom support regarding these problems.

**Comment by appzer** [comment]
Hi,

today i bought a homey on amazon! After arriveing i will check this and fix the issues.

**Comment by appzer** [comment]
Hi again,

today my Homey arrived! And all looks like it should! I do not see any problems in iOS app and on https://flow.homey.app/

![Foto 05 06 19, 13 03 13](https://user-images.githubusercontent.com/3443312/58957931-03388c00-87a2-11e9-866c-c81176d2d671.png)
![Foto 05 06 19, 13 03 21](https://user-images.githubusercontent.com/3443312/58957932-03d12280-87a2-11e9-9e78-47e939e62641.png)
![homey-flow-browser](https://user-images.githubusercontent.com/3443312/58957934-03d12280-87a2-11e9-80a7-4da4f9e87153.jpg)

**Comment by appzer** [comment]
is a little stupid > they use the field type as field labels from the app.json file!

**Comment by Veldkornet** [comment]
Yes, I spoke to support about it, they told me to fill out a [feature request](https://goo.gl/forms/nih5Hc6vkF0uzWt13)

**Comment by appzer** [comment]
did you do it

**Comment by appzer** [comment]
do you have the same issue with your pushover app

**Comment by Veldkornet** [comment]
Yes I did. But you should too or they won’t think it’s too important. This regarding the Field labels showing as text text number drop down number, but it occurs with all apps/flow, it’s not specific to PushSafer. 

I also did another feature request to add the field titles into the mobile app as well. Again though, you should too if you want to see that. 

For the sample text missing from the fields with the “number” data type, I spoke with support and they will look into it. 

Overall though, for the layout in the app vs the web, mine looks the same as yours.

**Comment by appzer** [comment]
i also send a feature request!
But Homey is a cool device, i have to say

**Comment by Veldkornet** [comment]
I've figured out that you cal just add the "title" field, to add the title (who would have thought).
Although it wasn't really documented.

I also saw that some of the default apps have some helpful text underneath (for example, delay, which is built into all the flows). I'm still trying to figure this one out though.

So for example:

```
{
  "name":"retry",
  "type":"number",
  "title": {
	"en":"Retry",
	"de":"Erneut",
	"nl":"Opnieuw"
  },
  "placeholder":{
	"en":"Retry 60-1080 or 0 = deactivated",
	"de":"Erneut senden 60-10800 oder 0 = deaktivert",
	"nl":"opnieuw versturen 60-10800 of 0 = gedeactiveerd"
  }
}
```

**Comment by appzer** [comment]
Great, i will try that

**Comment by Veldkornet** [comment]
Apparently, you can use “hint” with the relevant language tags which puts an extra line of information underneath the field, similar to how “title” is done. 

Edit: It seems as though the “hint” is for the entire flow action, and not for the individual arguments, in order to provide some additional information for the entire card... and that works fine. 

For the individual arguments, I’ll let you know when I figure it out since it’s not documented. 
However, the outcome for the individual arguments should look something like this:

![F9442CC6-3CEC-4AD3-B304-8B7A72C9B418](https://user-images.githubusercontent.com/10297361/59449122-ca825d80-8e06-11e9-9149-9bbaa6d61a88.jpeg)

**Comment by appzer** [comment]
@Veldkornet all works fine now, the labels are shown and the delay field is also visible and works like it should
